<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>API · BioMakie.jl</title><meta name="title" content="API · BioMakie.jl"/><meta property="og:title" content="API · BioMakie.jl"/><meta property="twitter:title" content="API · BioMakie.jl"/><meta name="description" content="Documentation for BioMakie.jl."/><meta property="og:description" content="Documentation for BioMakie.jl."/><meta property="twitter:description" content="Documentation for BioMakie.jl."/><meta property="og:url" content="https://kool7d.github.io/BioMakie.jl/API/"/><meta property="twitter:url" content="https://kool7d.github.io/BioMakie.jl/API/"/><link rel="canonical" href="https://kool7d.github.io/BioMakie.jl/API/"/><script data-outdated-warner src="../assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.048/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.3.0/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.4/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL=".."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="../assets/documenter.js"></script><script src="../siteinfo.js"></script><script src="../../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="../assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href="../">BioMakie.jl</a></span></div><form class="docs-search" action="../search/"><input class="docs-search-query" id="documenter-search-query" name="q" type="text" placeholder="Search docs (Ctrl + /)"/></form><ul class="docs-menu"><li><a class="tocitem" href="../">Home</a></li><li><a class="tocitem" href="../usage/">Usage</a></li><li><a class="tocitem" href="../Examples/alphashape/">Examples</a></li><li class="is-active"><a class="tocitem" href>API</a></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><div class="docs-left"><a class="docs-sidebar-button docs-navbar-link fa-solid fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a></div><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>API</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>API</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/kool7d/BioMakie.jl/blob/master/docs/src/API.md#" title="Edit source on GitHub"><span class="docs-icon fa-solid"></span></a><a class="docs-settings-button docs-navbar-link fa-solid fa-gear" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-article-toggle-button fa-solid fa-chevron-up" id="documenter-article-toggle-button" href="javascript:;" title="Collapse all docstrings"></a></div></header><article class="content" id="documenter-page"><h1 id="BioMakie-API"><a class="docs-heading-anchor" href="#BioMakie-API">BioMakie API</a><a id="BioMakie-API-1"></a><a class="docs-heading-anchor-permalink" href="#BioMakie-API" title="Permalink"></a></h1><ul><li><a href="#BioMakie.atomcolors-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.atomcolors</code></a></li><li><a href="#BioMakie.atomradii-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.atomradii</code></a></li><li><a href="#BioMakie.atomradius-Tuple{T} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.atomradius</code></a></li><li><a href="#BioMakie.atomsizes-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.atomsizes</code></a></li><li><a href="#BioMakie.backbonebonds-Tuple{BioStructures.Chain}"><code>BioMakie.backbonebonds</code></a></li><li><a href="#BioMakie.bondshape-Union{Tuple{Tuple{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.bondshape</code></a></li><li><a href="#BioMakie.bondshapes-Tuple{BioStructures.Chain}"><code>BioMakie.bondshapes</code></a></li><li><a href="#BioMakie.covalentbonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.covalentbonds</code></a></li><li><a href="#BioMakie.distancebonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.distancebonds</code></a></li><li><a href="#BioMakie.firstlabel-Tuple{Function}"><code>BioMakie.firstlabel</code></a></li><li><a href="#BioMakie.getbonds-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.getbonds</code></a></li><li><a href="#BioMakie.getbonds-Tuple{BioStructures.Chain, Vararg{Any}}"><code>BioMakie.getbonds</code></a></li><li><a href="#BioMakie.getinspectorlabel-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.getinspectorlabel</code></a></li><li><a href="#BioMakie.msavalues"><code>BioMakie.msavalues</code></a></li><li><a href="#BioMakie.plotmsa-Tuple{Any}"><code>BioMakie.plotmsa</code></a></li><li><a href="#BioMakie.plotmsa!-Union{Tuple{T}, Tuple{Figure, T}} where T&lt;:Union{Vector{Tuple{String, String}}, Vector{FASTX.FASTA.Record}, MIToS.MSA.AbstractMultipleSequenceAlignment}"><code>BioMakie.plotmsa!</code></a></li><li><a href="#BioMakie.plotstruc-Tuple{Any}"><code>BioMakie.plotstruc</code></a></li><li><a href="#BioMakie.plotstruc!-Union{Tuple{T}, Tuple{Figure, T}} where T&lt;:Union{Vector{MIToS.PDB.PDBAtom}, Vector{MIToS.PDB.PDBResidue}, OrderedCollections.OrderedDict, BioStructures.StructuralElementOrList}"><code>BioMakie.plotstruc!</code></a></li><li><a href="#BioMakie.plottingdata-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.plottingdata</code></a></li><li><a href="#BioMakie.plottingdata-Tuple{MIToS.MSA.AbstractMultipleSequenceAlignment}"><code>BioMakie.plottingdata</code></a></li><li><a href="#BioMakie.rescolors-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.rescolors</code></a></li><li><a href="#BioMakie.sidechainbonds-Tuple{BioStructures.AbstractResidue, Vararg{Any}}"><code>BioMakie.sidechainbonds</code></a></li><li><a href="#MakieCore.heatmap-Tuple{NamedArrays.NamedMatrix{Float64, PairwiseListMatrices.PairwiseListMatrix{Float64, false, Vector{Float64}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}"><code>MakieCore.heatmap</code></a></li><li><a href="#MakieCore.heatmap-Tuple{BioStructures.ContactMap}"><code>MakieCore.heatmap</code></a></li><li><a href="#MakieCore.heatmap-Tuple{NamedArrays.NamedMatrix{Bool, PairwiseListMatrices.PairwiseListMatrix{Bool, false, Vector{Bool}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}"><code>MakieCore.heatmap</code></a></li><li><a href="#MakieCore.heatmap-Tuple{BioStructures.DistanceMap}"><code>MakieCore.heatmap</code></a></li><li><a href="#MakieCore.heatmap!-Tuple{Figure, BioStructures.DistanceMap}"><code>MakieCore.heatmap!</code></a></li><li><a href="#MakieCore.heatmap!-Tuple{Figure, BioStructures.ContactMap}"><code>MakieCore.heatmap!</code></a></li><li><a href="#MakieCore.heatmap!-Tuple{Figure, NamedArrays.NamedMatrix{Bool, PairwiseListMatrices.PairwiseListMatrix{Bool, false, Vector{Bool}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}"><code>MakieCore.heatmap!</code></a></li><li><a href="#MakieCore.heatmap!-Tuple{Figure, NamedArrays.NamedMatrix{Float64, PairwiseListMatrices.PairwiseListMatrix{Float64, false, Vector{Float64}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}"><code>MakieCore.heatmap!</code></a></li></ul><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.atomcolors-Tuple{BioStructures.StructuralElementOrList}" href="#BioMakie.atomcolors-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.atomcolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomcolors( atoms )</code></pre><p>Get a Vector of colors for the atoms. To see all default element and amino acid colorschemes, use <code>getbiocolors()</code>. Keyword argument <code>colors</code> takes a Dict which maps element to color. (&quot;C&quot; =&gt; :red)</p><p>This function uses &#39;MIToS.PDB.bestoccupancy&#39; or &#39;defaultatom&#39; to ensure only one position per atom.</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/structure.jl#L198-L209">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.atomradii-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom" href="#BioMakie.atomradii-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.atomradii</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomradii( atoms )</code></pre><p>Collect atom radii based on element for plotting.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :ballandstick | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick, :spacefilling</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/structure.jl#L12-L19">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.atomradius-Tuple{T} where T&lt;:BioStructures.AbstractAtom" href="#BioMakie.atomradius-Tuple{T} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.atomradius</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomradius( atom )</code></pre><p>Collect atom radius based on element for plotting.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :ballandstick | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick, :spacefilling</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/structure.jl#L45-L52">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.atomsizes-Tuple{BioStructures.StructuralElementOrList}" href="#BioMakie.atomsizes-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.atomsizes</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomsizes( atms )</code></pre><p>Get a Vector of sizes for the atoms from a BioStructures.StructuralElementOrList.</p><p>This function uses &#39;MIToS.PDB.bestoccupancy&#39; or &#39;defaultatom&#39; to ensure only one position per atom.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :ballandstick | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick, :spacefilling</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/structure.jl#L278-L287">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.backbonebonds-Tuple{BioStructures.Chain}" href="#BioMakie.backbonebonds-Tuple{BioStructures.Chain}"><code>BioMakie.backbonebonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">backbonebonds( chn::BioStructures.Chain ) -&gt; BitMatrix</code></pre><p>Returns a matrix of backbone bonds in <code>chn</code>, where Mat[i,j] = 1 if atoms i and j are bonded. </p><p><strong>Keyword Arguments:</strong></p><ul><li>cutoff –––––- 1.6		# distance cutoff for bonds</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/bonds.jl#L375-L382">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.bondshape-Union{Tuple{Tuple{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom" href="#BioMakie.bondshape-Union{Tuple{Tuple{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.bondshape</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">bondshape( twoatoms )
bondshape( twopoints )</code></pre><p>Returns a (mesh) cylinder between two atoms or atomic coordinates.</p><p><strong>Keyword Arguments:</strong></p><ul><li>bondwidth ––––––- 0.2</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/bonds.jl#L984-L992">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.bondshapes-Tuple{BioStructures.Chain}" href="#BioMakie.bondshapes-Tuple{BioStructures.Chain}"><code>BioMakie.bondshapes</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">bondshapes( structure )
bondshapes( residues )
bondshapes( coordinates )
bondshapes( structure, bondmatrix )
bondshapes( residues, bondmatrix )
bondshapes( coordinates, bondmatrix )</code></pre><p>Returns a (mesh) cylinder between two atoms or points.</p><p><strong>Keyword Arguments:</strong></p><ul><li>algo ––––––––– :knowledgebased | :distance, :covalent	# unless bondmatrix is given</li><li>distance ––––––– 1.9										# unless bondmatrix is given</li><li>bondwidth ––––––- 0.2</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/bonds.jl#L1028-L1042">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.covalentbonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom" href="#BioMakie.covalentbonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.covalentbonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">covalentbonds( atms ) -&gt; BitMatrix</code></pre><p>Returns a matrix of all bonds in <code>atms</code>, where Mat[i,j] = 1 if atoms i and j are bonded. </p><p>This function uses &#39;bestoccupancy&#39; or &#39;defaultatom&#39; to ensure only one position per atom.</p><p><strong>Keyword Arguments:</strong></p><ul><li>extradistance –– 0.14  # fudge factor for better inclusion</li><li>H –––––––– true  # include bonds with hydrogen atoms</li><li>disulfides –––- false # include disulfide bonds</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/bonds.jl#L154-L165">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.distancebonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom" href="#BioMakie.distancebonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.distancebonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">distancebonds( atms ) -&gt; BitMatrix</code></pre><p>Returns a matrix of all bonds in <code>atms</code>, where Mat[i,j] = 1 if atoms i and j are bonded. </p><p>This function uses &#39;bestoccupancy&#39; or &#39;defaultatom&#39; to ensure only one position per atom.</p><p><strong>Keyword Arguments:</strong></p><ul><li>cutoff –––––- 1.9   # distance cutoff for bonds between heavy atoms</li><li>hydrogencutoff –- 1.14  # distance cutoff for bonds with hydrogen atoms</li><li>H –––––––– true  # include bonds with hydrogen atoms</li><li>disulfides –––- false # include disulfide bonds</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/bonds.jl#L9-L21">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.firstlabel-Tuple{Function}" href="#BioMakie.firstlabel-Tuple{Function}"><code>BioMakie.firstlabel</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">firstlabel( inspectorfunc::Function )
firstlabel( inspectorfunc::Observable{T} ) where {T&lt;:Function}</code></pre><p>Show an example of the inspector label function looks like. The position <code>p</code> will not be available to this function, so it will be set to <code>nothing</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/structure.jl#L182-L188">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.getbonds-Tuple{BioStructures.Chain, Vararg{Any}}" href="#BioMakie.getbonds-Tuple{BioStructures.Chain, Vararg{Any}}"><code>BioMakie.getbonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">getbonds( chn::BioStructures.Chain, selectors... ) -&gt; BitMatrix
getbonds( modl::BioStructures.Model, selectors... ) -&gt; BitMatrix
getbonds( struc::BioStructures.ProteinStructure, selectors... ) -&gt; BitMatrix</code></pre><p>Returns a matrix of all bonds in <code>chn</code>, where Mat[i,j] = 1 if atoms i and j are bonded. </p><p>This function uses &#39;bestoccupancy&#39; or &#39;defaultatom&#39; to ensure only one position per atom.</p><p><strong>Keyword Arguments:</strong></p><ul><li>algo ––––––- :knowledgebased 	# (:distance, :covalent) algorithm to find bonds</li><li>H –––––––– true				# include bonds with hydrogen atoms</li><li>cutoff –––––- 1.9				# distance cutoff for bonds between heavy atoms</li><li>extradistance –– 0.14				# fudge factor for better inclusion</li><li>disulfides –––- false				# include disulfide bonds</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/bonds.jl#L403-L418">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.getbonds-Tuple{Vector{MIToS.PDB.PDBResidue}}" href="#BioMakie.getbonds-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.getbonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">getbonds( residues ) -&gt; BitMatrix</code></pre><p>Returns a matrix of all bonds in <code>residues</code>, where Mat[i,j] = 1 if atoms i and j are bonded.</p><p><strong>Keyword Arguments:</strong></p><ul><li>algo ––––––- :knowledgebased 	# (:distance, :covalent) algorithm to find bonds</li><li>H –––––––– true				# include bonds with hydrogen atoms</li><li>cutoff –––––- 1.9				# distance cutoff for bonds between heavy atoms</li><li>extradistance –– 0.14				# fudge factor for better inclusion</li><li>disulfides –––- false				# include disulfide bonds</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/bonds.jl#L684-L695">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.getinspectorlabel-Tuple{BioStructures.StructuralElementOrList}" href="#BioMakie.getinspectorlabel-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.getinspectorlabel</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">getinspectorlabel( structure )
getinspectorlabel( residues )
getinspectorlabel( atom )</code></pre><p>Get the inspector label function for plotting a &#39;StructuralElementOrList&#39;.</p><p>This function uses &#39;MIToS.PDB.bestoccupancy&#39; or &#39;defaultatom&#39; to ensure only one position per atom.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/structure.jl#L78-L86">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.msavalues" href="#BioMakie.msavalues"><code>BioMakie.msavalues</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">msavalues( msa::AbstractMatrix, resdict::AbstractDict )::Matrix{Real}</code></pre><p>Returns a matrix of numbers according to the given dictionary, where keys are residue letters and values are numbers. This matrix is used as input for <code>plotmsa</code> for the heatmap colors.</p><p>Default values for residue letters are from Kidera Factor values.  kf 2 is Kidera Factor 2 (size/volume-related). The KF dictionary is in <code>utils.jl</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/msa.jl#L96-L105">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.plotmsa!-Union{Tuple{T}, Tuple{Figure, T}} where T&lt;:Union{Vector{Tuple{String, String}}, Vector{FASTX.FASTA.Record}, MIToS.MSA.AbstractMultipleSequenceAlignment}" href="#BioMakie.plotmsa!-Union{Tuple{T}, Tuple{Figure, T}} where T&lt;:Union{Vector{Tuple{String, String}}, Vector{FASTX.FASTA.Record}, MIToS.MSA.AbstractMultipleSequenceAlignment}"><code>BioMakie.plotmsa!</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotmsa!( fig, msa )</code></pre><p>Plot a multiple sequence alignment (MSA) into a Figure. </p><p><strong>Example</strong></p><pre><code class="language-julia hljs">fig = Figure(resolution = (1100, 400))

plotmsa!( fig::Figure, msa::T; kwargs... ) where {T&lt;:Union{MSA.AbstractMultipleSequenceAlignment,
											   Vector{Tuple{String,String}},
											   Vector{FASTX.FASTA.Record}}}</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>xlabels –––- {1:height}</li><li>ylabels –––- {1:width}</li><li>sheetsize ––- [40,20]</li><li>gridposition – (1,1)</li><li>markersize –– 12</li><li>colorscheme –- :buda</li><li>markercolor –- :black</li><li>kwargs...   					# forwarded to scatter plot</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/msa.jl#L118-L141">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.plotmsa-Tuple{Any}" href="#BioMakie.plotmsa-Tuple{Any}"><code>BioMakie.plotmsa</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotmsa( msa )
plotmsa( plotdata )</code></pre><p>Plot a multiple sequence alignment (MSA). Returns a Figure, or a Figure and Observables for interaction. </p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">downloadpfam(&quot;PF00062&quot;)
msa = MIToS.MSA.read(&quot;PF00062.stockholm.gz&quot;, Stockholm, 
					generatemapping =true, useidcoordinates=true)

plotmsa( msa; kwargs... )</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>resolution –––– (1100, 400)</li><li>sheetsize ––––- [40,20]</li><li>gridposition ––– (1,1)</li><li>colorscheme –––- :viridis</li><li>resdict –––––- kideradict    # Dictionary of values (::Dict{String,Float}, &quot;Y&quot; =&gt; 1.48) for heatmap.</li><li>kf –––––––– 2             # If resdict == kideradict, this is the Kidera Factor. KF2 is size/volume-related.</li><li>kwargs...    						# forwarded to scatter plot</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/msa.jl#L573-L597">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.plotstruc!-Union{Tuple{T}, Tuple{Figure, T}} where T&lt;:Union{Vector{MIToS.PDB.PDBAtom}, Vector{MIToS.PDB.PDBResidue}, OrderedCollections.OrderedDict, BioStructures.StructuralElementOrList}" href="#BioMakie.plotstruc!-Union{Tuple{T}, Tuple{Figure, T}} where T&lt;:Union{Vector{MIToS.PDB.PDBAtom}, Vector{MIToS.PDB.PDBResidue}, OrderedCollections.OrderedDict, BioStructures.StructuralElementOrList}"><code>BioMakie.plotstruc!</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotstruc!( fig, structure )
plotstruc!( gridposition, structure )
plotstruc!( fig, plotdata )
plotstruc!( gridposition, plotdata )</code></pre><p>Plot a protein structure(/chain/residues/atoms) into a Figure. </p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">fig = Figure()

using MIToS.PDB

pdbfile = MIToS.PDB.downloadpdb(&quot;2vb1&quot;)
struc = MIToS.PDB.read(pdbfile, PDBML) |&gt; Observable
strucplot = plotstruc!(fig, struc)

chain_A = @residues struc model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All
strucplot = plotstruc!(fig, chain_A)

chnatms = @atoms struc model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All atom All
strucplot = plotstruc!(fig, chnatms)
-------------------------
using BioStructures

struc = retrievepdb(&quot;2vb1&quot;, dir = &quot;data/&quot;) |&gt; Observable
strucplot = plotstruc!(fig, struc)

struc = read(&quot;data/2vb1_mutant1.pdb&quot;, BioStructures.PDB) |&gt; Observable
strucplot = plotstruc!(fig, struc)

chain_A = retrievepdb(&quot;2hhb&quot;, dir = &quot;data/&quot;)[&quot;A&quot;] |&gt; Observable
strucplot = plotstruc!(fig, chain_A)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>resolution ––- (800,600)</li><li>gridposition –- (1,1)  # if an MSA is already plotted, (2,1:3) works well</li><li>plottype –––- :ballandstick, :covalent, or :spacefilling</li><li>atomcolors ––- elecolors, others in <code>getbiocolors()</code>, or provide a Dict like: &quot;N&quot; =&gt; :blue</li><li>markersize ––- 0.0</li><li>markerscale –– 1.0</li><li>bondtype –––- :knowledgebased, :covalent, or :distance</li><li>distance –––- 1.9  # distance cutoff for covalent bonds</li><li>inspectorlabel - :default, or define your own function like: (self, i, p) -&gt; &quot;atom: ... coords: ...&quot;</li><li>water ––––– false  # show water molecules</li><li>kwargs... ––– keyword arguments passed to the atom <code>meshscatter</code></li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/structure.jl#L469-L517">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.plotstruc-Tuple{Any}" href="#BioMakie.plotstruc-Tuple{Any}"><code>BioMakie.plotstruc</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotstruc( structure )
plotstruc( residues )
plotstruc( plotdata )</code></pre><p>Create and return a Makie Figure for a protein structural element. </p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">using MIToS.PDB

pdbfile = MIToS.PDB.downloadpdb(&quot;2vb1&quot;)
struc = MIToS.PDB.read(pdbfile, PDBML) |&gt; Observable
strucplot = plotstruc(struc)

chain_A = @residues struc model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All
strucplot = plotstruc(chain_A)

chnatms = @atoms struc model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All atom All
strucplot = plotstruc(chnatms)
-------------------------
using BioStructures

struc = retrievepdb(&quot;2vb1&quot;, dir = &quot;data/&quot;) |&gt; Observable
strucplot = plotstruc(struc)

struc = read(&quot;data/2vb1_mutant1.pdb&quot;, BioStructures.PDB) |&gt; Observable
strucplot = plotstruc(struc)

chain_A = retrievepdb(&quot;2hhb&quot;, dir = &quot;data/&quot;)[&quot;A&quot;] |&gt; Observable
strucplot = plotstruc(chain_A)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>resolution ––- (800,600)</li><li>gridposition –- (1,1)  # if an MSA is already plotted, (2,1:3) works well</li><li>plottype –––- :ballandstick, :covalent, or :spacefilling</li><li>atomcolors ––- elecolors, others in <code>getbiocolors()</code>, or provide a Dict like: &quot;N&quot; =&gt; :blue</li><li>markersize ––- 0.0</li><li>markerscale –– 1.0</li><li>bondtype –––- :knowledgebased, :covalent, or :distance</li><li>distance –––- 1.9  # distance cutoff for covalent bonds</li><li>inspectorlabel - :default, or define your own function like: (self, i, p) -&gt; &quot;atom: ... coords: ...&quot;</li><li>water ––––– false  # show water molecules</li><li>kwargs... ––– keyword arguments passed to the atom <code>meshscatter</code></li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/structure.jl#L824-L869">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.plottingdata-Tuple{BioStructures.StructuralElementOrList}" href="#BioMakie.plottingdata-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.plottingdata</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plottingdata( structure )
plottingdata( residues )
plottingdata( atoms )</code></pre><p>This function returns an OrderedDict of the main data used for plotting.  This function uses &#39;MIToS.PDB.bestoccupancy&#39; or &#39;defaultatom&#39; to ensure only one position per atom. By default the kwarg &#39;water&#39; is set to false, so water molecules are not included.</p><p><strong>Returns:</strong></p><pre><code class="nohighlight hljs">OrderedDict(&quot;atoms&quot; =&gt; ..., 
            &quot;coords&quot; =&gt; ..., 
            &quot;colors&quot; =&gt; ...,
            &quot;sizes&quot; =&gt; ...,
            &quot;bonds&quot; =&gt; ...)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>colors –––- elecolors      | Options - elecolors, aquacolors, shapelycolors, maecolors</li><li>radiustype –- :ballandstick  | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick, :spacefilling</li><li>water –––– false          | Options - true, false</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/structure.jl#L317-L337">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.plottingdata-Tuple{MIToS.MSA.AbstractMultipleSequenceAlignment}" href="#BioMakie.plottingdata-Tuple{MIToS.MSA.AbstractMultipleSequenceAlignment}"><code>BioMakie.plottingdata</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plottingdata( msa )</code></pre><p>Collects data for plotting (residue string matrix, matrix heatmap values,  x labels, and y labels) from a multiple sequence alignment (MSA) object. </p><p>The MSA object can be a: </p><ul><li><code>AbstractMultipleSequenceAlignment</code> from MIToS.MSA, </li><li>vector of tuples &#39;Vector{Tuple{String,String}}&#39; from FastaIO, </li><li>vector of FASTA records &#39;Vector{FASTX.FASTA.Record}&#39; from FASTX.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/msa.jl#L6-L16">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.rescolors-Tuple{BioStructures.StructuralElementOrList}" href="#BioMakie.rescolors-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.rescolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">rescolors( residues )</code></pre><p>Get a Vector of colors for the atoms. To see all default element and amino acid colorschemes, use <code>getbiocolors()</code>. Keyword argument <code>colors</code> takes a Dict which maps residue to color. (&quot;C&quot; =&gt; :red)</p><p>This function uses &#39;MIToS.PDB.bestoccupancy&#39; or &#39;defaultatom&#39; to ensure only one position per atom.</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors, shapelycolors, maecolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/structure.jl#L239-L250">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="BioMakie.sidechainbonds-Tuple{BioStructures.AbstractResidue, Vararg{Any}}" href="#BioMakie.sidechainbonds-Tuple{BioStructures.AbstractResidue, Vararg{Any}}"><code>BioMakie.sidechainbonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">sidechainbonds( res::BioStructures.AbstractResidue, selectors... ) -&gt; BitMatrix</code></pre><p>Returns a matrix of sidechain bonds in <code>res</code>, where Mat[i,j] = 1 if atoms i and j are bonded.</p><p>This function uses &#39;bestoccupancy&#39; or &#39;defaultatom&#39; to ensure only one position per atom.</p><p><strong>Keyword Arguments:</strong></p><ul><li>algo ––––––- :knowledgebased 	# (:distance, :covalent) algorithm to find bonds</li><li>H –––––––– true				# include bonds with hydrogen atoms</li><li>cutoff –––––- 1.9				# distance cutoff for bonds between heavy atoms</li><li>extradistance –– 0.14				# fudge factor for better inclusion</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/bonds.jl#L290-L302">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="MakieCore.heatmap!-Tuple{Figure, BioStructures.ContactMap}" href="#MakieCore.heatmap!-Tuple{Figure, BioStructures.ContactMap}"><code>MakieCore.heatmap!</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">heatmap!( fig, cmap; kwargs... )</code></pre><p>Plot a BioStructures contact map.</p><p><strong>Example</strong></p><pre><code class="language-julia hljs">fig = Figure()

using BioStructures

struc = retrievepdb(&quot;1IVO&quot;)[1]
cbetas_A = collectatoms(struc[&quot;A&quot;], cbetaselector)
cbetas_B = collectatoms(struc[&quot;B&quot;], cbetaselector)
cmap = ContactMap(cbetas_A, cbetas_B)
heatmap!(fig, cmap)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>xlabel ––––––––- &quot;Item 2&quot;</li><li>ylabel ––––––––- &quot;Item 1&quot;</li><li>colormap –––––––- :ice</li><li>kwargs... ––––––– Keyword arguments to pass to heatmap</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/otherplots.jl#L325-L349">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="MakieCore.heatmap!-Tuple{Figure, BioStructures.DistanceMap}" href="#MakieCore.heatmap!-Tuple{Figure, BioStructures.DistanceMap}"><code>MakieCore.heatmap!</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">heatmap( dmap; kwargs... )</code></pre><p>Plot a BioStructures distance map.</p><p><strong>Example</strong></p><pre><code class="language-julia hljs">using BioStructures

struc = retrievepdb(&quot;1IVO&quot;)[1]
cbetas_A = collectatoms(struc[&quot;A&quot;], cbetaselector)
cbetas_B = collectatoms(struc[&quot;B&quot;], cbetaselector)
dmap = DistanceMap(cbetas_A, cbetas_B)
heatmap(dmap)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>xlabel ––––––––- &quot;Item 2&quot;</li><li>ylabel ––––––––- &quot;Item 1&quot;</li><li>colormap –––––––- :viridis</li><li>kwargs... ––––––– additional keyword arguments to pass to heatmap</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/otherplots.jl#L243-L265">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="MakieCore.heatmap!-Tuple{Figure, NamedArrays.NamedMatrix{Bool, PairwiseListMatrices.PairwiseListMatrix{Bool, false, Vector{Bool}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}" href="#MakieCore.heatmap!-Tuple{Figure, NamedArrays.NamedMatrix{Bool, PairwiseListMatrices.PairwiseListMatrix{Bool, false, Vector{Bool}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}"><code>MakieCore.heatmap!</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">heatmap!( fig, cmap; kwargs... )</code></pre><p>Plot a MIToS contact map.</p><p>Example:</p><pre><code class="language-julia hljs">fig = Figure()

using MIToS.PDB

pdbfile = MIToS.PDB.downloadpdb(&quot;1IVO&quot;, format=PDBFile)
residues_1ivo = read(pdbfile, PDBFile)
pdb = @residues residues_1ivo model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All
cmap = contact(pdb, 8.0, criteria=&quot;CB&quot;)

heatmap!(fig, cmap)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>xlabel ––––––––- &quot;Item 2&quot;</li><li>ylabel ––––––––- &quot;Item 1&quot;</li><li>colormap –––––––- :ice</li><li>kwargs... ––––––– additional keyword arguments to pass to heatmap</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/otherplots.jl#L153-L178">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="MakieCore.heatmap!-Tuple{Figure, NamedArrays.NamedMatrix{Float64, PairwiseListMatrices.PairwiseListMatrix{Float64, false, Vector{Float64}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}" href="#MakieCore.heatmap!-Tuple{Figure, NamedArrays.NamedMatrix{Float64, PairwiseListMatrices.PairwiseListMatrix{Float64, false, Vector{Float64}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}"><code>MakieCore.heatmap!</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">heatmap!( fig, dmap; kwargs... )</code></pre><p>Plot a MIToS distance map.</p><p><strong>Example</strong></p><pre><code class="language-julia hljs">fig = Figure()

using MIToS.PDB

pdbfile = MIToS.PDB.downloadpdb(&quot;1IVO&quot;, format=PDBFile)
residues_1ivo = read(pdbfile, PDBFile)
pdb = @residues residues_1ivo model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All
dmap = MIToS.PDB.distance(pdb, criteria=&quot;All&quot;)

heatmap!(fig, dmap)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>xlabel ––––––––- &quot;Item 2&quot;</li><li>ylabel ––––––––- &quot;Item 1&quot;</li><li>colormap –––––––- :viridis</li><li>kwargs... ––––––– additional keyword arguments to pass to heatmap</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/otherplots.jl#L57-L82">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="MakieCore.heatmap-Tuple{BioStructures.ContactMap}" href="#MakieCore.heatmap-Tuple{BioStructures.ContactMap}"><code>MakieCore.heatmap</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">heatmap( cmap; kwargs... )</code></pre><p>Plot a BioStructures contact map.</p><p><strong>Example</strong></p><pre><code class="language-julia hljs">using BioStructures

struc = retrievepdb(&quot;1IVO&quot;)[1]
cbetas_A = collectatoms(struc[&quot;A&quot;], cbetaselector)
cbetas_B = collectatoms(struc[&quot;B&quot;], cbetaselector)
cmap = ContactMap(cbetas_A, cbetas_B)
heatmap(cmap)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>xlabel ––––––––- &quot;Item 2&quot;</li><li>ylabel ––––––––- &quot;Item 1&quot;</li><li>colormap –––––––- :ice</li><li>kwargs... ––––––– additional keyword arguments to pass to heatmap</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/otherplots.jl#L284-L306">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="MakieCore.heatmap-Tuple{BioStructures.DistanceMap}" href="#MakieCore.heatmap-Tuple{BioStructures.DistanceMap}"><code>MakieCore.heatmap</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">heatmap( dmap; kwargs... )</code></pre><p>Plot a BioStructures distance map.</p><p><strong>Example</strong></p><pre><code class="language-julia hljs">using BioStructures

struc = retrievepdb(&quot;1IVO&quot;)[1]
cbetas_A = collectatoms(struc[&quot;A&quot;], cbetaselector)
cbetas_B = collectatoms(struc[&quot;B&quot;], cbetaselector)
dmap = DistanceMap(cbetas_A, cbetas_B)
heatmap(dmap)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>xlabel ––––––––- &quot;Item 2&quot;</li><li>ylabel ––––––––- &quot;Item 1&quot;</li><li>colormap –––––––- :viridis</li><li>kwargs... ––––––– additional keyword arguments to pass to heatmap</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/otherplots.jl#L201-L223">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="MakieCore.heatmap-Tuple{NamedArrays.NamedMatrix{Bool, PairwiseListMatrices.PairwiseListMatrix{Bool, false, Vector{Bool}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}" href="#MakieCore.heatmap-Tuple{NamedArrays.NamedMatrix{Bool, PairwiseListMatrices.PairwiseListMatrix{Bool, false, Vector{Bool}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}"><code>MakieCore.heatmap</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">heatmap( cmap; kwargs... )</code></pre><p>Plot a MIToS contact map.</p><p>Example:</p><pre><code class="language-julia hljs">using MIToS.PDB

pdbfile = MIToS.PDB.downloadpdb(&quot;1IVO&quot;, format=PDBFile)
residues_1ivo = read(pdbfile, PDBFile)
pdb = @residues residues_1ivo model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All
cmap = contact(pdb, 8.0, criteria=&quot;CB&quot;)

heatmap(cmap)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>xlabel ––––––––- &quot;Item 2&quot;</li><li>ylabel ––––––––- &quot;Item 1&quot;</li><li>colormap –––––––- Colormap to use</li><li>kwargs... ––––––– additional keyword arguments to pass to heatmap</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/otherplots.jl#L106-L129">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="MakieCore.heatmap-Tuple{NamedArrays.NamedMatrix{Float64, PairwiseListMatrices.PairwiseListMatrix{Float64, false, Vector{Float64}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}" href="#MakieCore.heatmap-Tuple{NamedArrays.NamedMatrix{Float64, PairwiseListMatrices.PairwiseListMatrix{Float64, false, Vector{Float64}}, Tuple{OrderedCollections.OrderedDict{String, Int64}, OrderedCollections.OrderedDict{String, Int64}}}}"><code>MakieCore.heatmap</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">heatmap( dmap; kwargs... )</code></pre><p>Plot a MIToS distance map.</p><p><strong>Example</strong></p><pre><code class="language-julia hljs">using MIToS.PDB

pdbfile = MIToS.PDB.downloadpdb(&quot;1IVO&quot;, format=PDBFile)
residues_1ivo = read(pdbfile, PDBFile)
pdb = @residues residues_1ivo model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All
dmap = MIToS.PDB.distance(pdb, criteria=&quot;All&quot;)

heatmap(dmap)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>xlabel ––––––––- &quot;Item 2&quot;</li><li>ylabel ––––––––- &quot;Item 1&quot;</li><li>colormap –––––––- :viridis</li><li>kwargs... ––––––– additional keyword arguments to pass to heatmap</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/c5376fae541fdb1d34bcd7a0d2fb1e117c0891d9/src/otherplots.jl#L9-L32">source</a></section></article></article><nav class="docs-footer"><a class="docs-footer-prevpage" href="../Examples/alphashape/">« Examples</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 0.28.0-DEV on <span class="colophon-date" title="Wednesday 3 May 2023 00:46">Wednesday 3 May 2023</span>. Using Julia version 1.8.5.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
