<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>API · BioMakie.jl</title><meta name="title" content="API · BioMakie.jl"/><meta property="og:title" content="API · BioMakie.jl"/><meta property="twitter:title" content="API · BioMakie.jl"/><meta name="description" content="Documentation for BioMakie.jl."/><meta property="og:description" content="Documentation for BioMakie.jl."/><meta property="twitter:description" content="Documentation for BioMakie.jl."/><meta property="og:url" content="https://kool7d.github.io/BioMakie.jl/API/"/><meta property="twitter:url" content="https://kool7d.github.io/BioMakie.jl/API/"/><link rel="canonical" href="https://kool7d.github.io/BioMakie.jl/API/"/><script data-outdated-warner src="../assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.045/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.24/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL=".."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="../assets/documenter.js"></script><script src="../siteinfo.js"></script><script src="../../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="../assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href="../">BioMakie.jl</a></span></div><form class="docs-search" action="../search/"><input class="docs-search-query" id="documenter-search-query" name="q" type="text" placeholder="Search docs (Ctrl + /)"/></form><ul class="docs-menu"><li><a class="tocitem" href="../">Home</a></li><li class="is-active"><a class="tocitem" href>API</a></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>API</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>API</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/kool7d/BioMakie.jl/blob/master/docs/src/API.md#" title="Edit source on GitHub"><span class="docs-icon fas"></span></a><a class="docs-settings-button docs-navbar-link fas fa-cog" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-sidebar-button docs-navbar-link fa fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a></div></header><article class="content" id="documenter-page"><h1 id="BioMakie-API"><a class="docs-heading-anchor" href="#BioMakie-API">BioMakie API</a><a id="BioMakie-API-1"></a><a class="docs-heading-anchor-permalink" href="#BioMakie-API" title="Permalink"></a></h1><ul><li><a href="#BioMakie.atomcolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}"><code>BioMakie.atomcolors</code></a></li><li><a href="#BioMakie.atomcolors-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.atomcolors</code></a></li><li><a href="#BioMakie.atomcolors-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.atomcolors</code></a></li><li><a href="#BioMakie.atomcolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}"><code>BioMakie.atomcolors</code></a></li><li><a href="#BioMakie.atomcolors-Tuple{Vector{MIToS.PDB.PDBAtom}}"><code>BioMakie.atomcolors</code></a></li><li><a href="#BioMakie.atomcolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList"><code>BioMakie.atomcolors</code></a></li><li><a href="#BioMakie.atomradii-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.atomradii</code></a></li><li><a href="#BioMakie.atomradii-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:MIToS.PDB.PDBAtom"><code>BioMakie.atomradii</code></a></li><li><a href="#BioMakie.atomradius-Tuple{T} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.atomradius</code></a></li><li><a href="#BioMakie.atomradius-Tuple{T} where T&lt;:MIToS.PDB.PDBAtom"><code>BioMakie.atomradius</code></a></li><li><a href="#BioMakie.atomsizes-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.atomsizes</code></a></li><li><a href="#BioMakie.atomsizes-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList"><code>BioMakie.atomsizes</code></a></li><li><a href="#BioMakie.atomsizes-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.atomsizes</code></a></li><li><a href="#BioMakie.atomsizes-Tuple{Vector{MIToS.PDB.PDBAtom}}"><code>BioMakie.atomsizes</code></a></li><li><a href="#BioMakie.atomsizes-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}"><code>BioMakie.atomsizes</code></a></li><li><a href="#BioMakie.atomsizes-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}"><code>BioMakie.atomsizes</code></a></li><li><a href="#BioMakie.backbonebonds-Tuple{BioStructures.Chain}"><code>BioMakie.backbonebonds</code></a></li><li><a href="#BioMakie.covalentbonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.covalentbonds</code></a></li><li><a href="#BioMakie.distancebonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.distancebonds</code></a></li><li><a href="#BioMakie.firstlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Function"><code>BioMakie.firstlabel</code></a></li><li><a href="#BioMakie.firstlabel-Tuple{Function}"><code>BioMakie.firstlabel</code></a></li><li><a href="#BioMakie.getbonds-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.getbonds</code></a></li><li><a href="#BioMakie.getbonds-Tuple{BioStructures.Chain, Vararg{Any}}"><code>BioMakie.getbonds</code></a></li><li><a href="#BioMakie.getbonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractResidue"><code>BioMakie.getbonds</code></a></li><li><a href="#BioMakie.inspectorlabel-Tuple{Vector{MIToS.PDB.PDBAtom}}"><code>BioMakie.inspectorlabel</code></a></li><li><a href="#BioMakie.inspectorlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList"><code>BioMakie.inspectorlabel</code></a></li><li><a href="#BioMakie.inspectorlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}"><code>BioMakie.inspectorlabel</code></a></li><li><a href="#BioMakie.inspectorlabel-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.inspectorlabel</code></a></li><li><a href="#BioMakie.inspectorlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}"><code>BioMakie.inspectorlabel</code></a></li><li><a href="#BioMakie.inspectorlabel-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.inspectorlabel</code></a></li><li><a href="#BioMakie.msavalues"><code>BioMakie.msavalues</code></a></li><li><a href="#BioMakie.plotmsa-Tuple{Any, Any}"><code>BioMakie.plotmsa</code></a></li><li><a href="#BioMakie.plotmsa-NTuple{4, Any}"><code>BioMakie.plotmsa</code></a></li><li><a href="#BioMakie.plotmsa-Tuple{Any}"><code>BioMakie.plotmsa</code></a></li><li><a href="#BioMakie.plotmsa!"><code>BioMakie.plotmsa!</code></a></li><li><a href="#BioMakie.plotmsa!-NTuple{4, Any}"><code>BioMakie.plotmsa!</code></a></li><li><a href="#BioMakie.plotmsa!-NTuple{5, Any}"><code>BioMakie.plotmsa!</code></a></li><li><a href="#BioMakie.plotstruc-Tuple{Observable}"><code>BioMakie.plotstruc</code></a></li><li><a href="#BioMakie.plotstruc-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.plotstruc</code></a></li><li><a href="#BioMakie.plotstruc!-Tuple{Figure, Observable}"><code>BioMakie.plotstruc!</code></a></li><li><a href="#BioMakie.plotstruc!-Tuple{Any, Any}"><code>BioMakie.plotstruc!</code></a></li><li><a href="#BioMakie.plottingdata-Tuple{MIToS.MSA.AbstractMultipleSequenceAlignment}"><code>BioMakie.plottingdata</code></a></li><li><a href="#BioMakie.plottingdata-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}"><code>BioMakie.plottingdata</code></a></li><li><a href="#BioMakie.plottingdata-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList"><code>BioMakie.plottingdata</code></a></li><li><a href="#BioMakie.plottingdata-Tuple{Vector{FASTX.FASTA.Record}}"><code>BioMakie.plottingdata</code></a></li><li><a href="#BioMakie.plottingdata-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}"><code>BioMakie.plottingdata</code></a></li><li><a href="#BioMakie.plottingdata-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.plottingdata</code></a></li><li><a href="#BioMakie.plottingdata-Tuple{Vector{Tuple{String, String}}}"><code>BioMakie.plottingdata</code></a></li><li><a href="#BioMakie.plottingdata-Tuple{Vector{MIToS.PDB.PDBAtom}}"><code>BioMakie.plottingdata</code></a></li><li><a href="#BioMakie.plottingdata-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.plottingdata</code></a></li><li><a href="#BioMakie.rescolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList"><code>BioMakie.rescolors</code></a></li><li><a href="#BioMakie.rescolors-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.rescolors</code></a></li><li><a href="#BioMakie.rescolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}"><code>BioMakie.rescolors</code></a></li><li><a href="#BioMakie.rescolors-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.rescolors</code></a></li><li><a href="#BioMakie.sidechainbonds-Tuple{BioStructures.AbstractResidue, Vararg{Any}}"><code>BioMakie.sidechainbonds</code></a></li></ul><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomcolors-Tuple{BioStructures.StructuralElementOrList}" href="#BioMakie.atomcolors-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.atomcolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomcolors( struc::BioStructures.StructuralElementOrList )</code></pre><p>Get a Vector of colors for the atoms in a BioStructures StructuralElementOrList. To see all default element and amino acid colorschemes, use the <code>getbiocolors()</code> function. Keyword argument <code>colors</code> takes a Dict which maps element to color. (&quot;C&quot; =&gt; :red)</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L192-L201">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomcolors-Tuple{Vector{MIToS.PDB.PDBAtom}}" href="#BioMakie.atomcolors-Tuple{Vector{MIToS.PDB.PDBAtom}}"><code>BioMakie.atomcolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomcolors( atms::Vector{MIToS.PDB.PDBAtom} )</code></pre><p>Get a Vector of colors for the atoms from a Vector of MIToS.PDB.PDBAtom. To see all default element and amino acid colorschemes, use the <code>getbiocolors()</code> function. Keyword argument <code>colors</code> takes a Dict which maps element to color. (&quot;C&quot; =&gt; :red)</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L224-L233">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomcolors-Tuple{Vector{MIToS.PDB.PDBResidue}}" href="#BioMakie.atomcolors-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.atomcolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomcolors( resz::Vector{MIToS.PDB.PDBResidue} )</code></pre><p>Get a Vector of colors for the atoms from a Vector of MIToS.PDB.PDBResidue. To see all default element and amino acid colorschemes, use the <code>getbiocolors()</code> function. Keyword argument <code>colors</code> takes a Dict which maps element to color. (&quot;C&quot; =&gt; :red)</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L208-L217">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomcolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList" href="#BioMakie.atomcolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList"><code>BioMakie.atomcolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomcolors( struc::Observable{T} ) where {T&lt;:BioStructures.StructuralElementOrList}</code></pre><p>Get a Vector of colors for the atoms in a BioStructures StructuralElementOrList. To see all default element and amino acid colorschemes, use the <code>getbiocolors()</code> function. Keyword argument <code>colors</code> takes a Dict which maps element to color. (&quot;C&quot; =&gt; :red)</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L239-L248">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomcolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}" href="#BioMakie.atomcolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}"><code>BioMakie.atomcolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomcolors( atms::Observable{T} ) where {T&lt;:Vector{MIToS.PDB.PDBAtom}}</code></pre><p>Get a Vector of colors for the atoms from a Vector of MIToS.PDB.PDBAtoms. To see all default element and amino acid colorschemes, use the <code>getbiocolors()</code> function. Keyword argument <code>colors</code> takes a Dict which maps element to color. (&quot;C&quot; =&gt; :red)</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L271-L280">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomcolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}" href="#BioMakie.atomcolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}"><code>BioMakie.atomcolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomcolors( resz::Observable{T} ) where {T&lt;:Vector{MIToS.PDB.PDBResidue}}</code></pre><p>Get a Vector of colors for the atoms from a Vector of MIToS.PDB.PDBResidues. To see all default element and amino acid colorschemes, use the <code>getbiocolors()</code> function. Keyword argument <code>colors</code> takes a Dict which maps element to color. (&quot;C&quot; =&gt; :red)</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L255-L264">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomradii-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom" href="#BioMakie.atomradii-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.atomradii</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomradii( atoms::Vector{T} ) where T&lt;:BioStructures.AbstractAtom</code></pre><p>Collect atom radii based on element for plotting, from a Vector of BioStructures.AbstractAtoms.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L6-L13">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomradii-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:MIToS.PDB.PDBAtom" href="#BioMakie.atomradii-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:MIToS.PDB.PDBAtom"><code>BioMakie.atomradii</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomradii( atoms::Vector{T} ) where T&lt;:MIToS.PDB.PDBAtom</code></pre><p>Collect atom radii based on element for plotting, from a Vector of MIToS.PDB.PDBAtoms.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L27-L34">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomradius-Tuple{T} where T&lt;:BioStructures.AbstractAtom" href="#BioMakie.atomradius-Tuple{T} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.atomradius</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomradius( atom::BioStructures.Atom )</code></pre><p>Collect atom radius based on element for plotting.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L48-L55">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomradius-Tuple{T} where T&lt;:MIToS.PDB.PDBAtom" href="#BioMakie.atomradius-Tuple{T} where T&lt;:MIToS.PDB.PDBAtom"><code>BioMakie.atomradius</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomradius( atom::MIToS.PDB.PDBAtom )</code></pre><p>Collect atom radius based on element for plotting.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L69-L76">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomsizes-Tuple{BioStructures.StructuralElementOrList}" href="#BioMakie.atomsizes-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.atomsizes</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomsizes( struc::BioStructures.StructuralElementOrList )</code></pre><p>Get a Vector of sizes for the atoms from a BioStructures.StructuralElementOrList.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L356-L363">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomsizes-Tuple{Vector{MIToS.PDB.PDBAtom}}" href="#BioMakie.atomsizes-Tuple{Vector{MIToS.PDB.PDBAtom}}"><code>BioMakie.atomsizes</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomsizes( atms::Vector{MIToS.PDB.PDBAtom} )</code></pre><p>Get a Vector of sizes for the atoms from a Vector of MIToS.PDB.PDBAtoms.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L384-L391">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomsizes-Tuple{Vector{MIToS.PDB.PDBResidue}}" href="#BioMakie.atomsizes-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.atomsizes</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomsizes( resz::Vector{MIToS.PDB.PDBResidue} )</code></pre><p>Get a Vector of sizes for the atoms from a Vector of MIToS.PDB.PDBResidues.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L370-L377">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomsizes-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList" href="#BioMakie.atomsizes-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList"><code>BioMakie.atomsizes</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomsizes( struc::Observable{T} ) where {T&lt;:BioStructures.StructuralElementOrList}</code></pre><p>Get a Vector of sizes for the atoms from a BioStructures.StructuralElementOrList.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L397-L404">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomsizes-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}" href="#BioMakie.atomsizes-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}"><code>BioMakie.atomsizes</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomsizes( atms::Observable{T} ) where {T&lt;:Vector{MIToS.PDB.PDBAtom}}</code></pre><p>Get a Vector of sizes for the atoms from a Vector of MIToS.PDB.PDBAtoms.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L425-L432">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.atomsizes-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}" href="#BioMakie.atomsizes-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}"><code>BioMakie.atomsizes</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">atomsizes( resz::Observable{T} ) where {T&lt;:Vector{MIToS.PDB.PDBResidue}}</code></pre><p>Get a Vector of sizes for the atoms from a Vector of MIToS.PDB.PDBResidues.</p><p><strong>Keyword Arguments:</strong></p><ul><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L411-L418">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.backbonebonds-Tuple{BioStructures.Chain}" href="#BioMakie.backbonebonds-Tuple{BioStructures.Chain}"><code>BioMakie.backbonebonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">backbonebonds( chn::BioStructures.Chain )</code></pre><p>Returns a matrix of backbone bonds in <code>chn</code>. </p><p><strong>Optional Arguments:</strong></p><ul><li>cutoff –––––- 1.6		# distance cutoff for bonds</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/bonds.jl#L366-L373">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.covalentbonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom" href="#BioMakie.covalentbonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.covalentbonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">covalentbonds( atms::Vector{T} )</code></pre><p>Returns a matrix of all bonds in <code>atms</code>. </p><p><strong>Optional Arguments:</strong></p><ul><li>extradistance –– 0.14  # fudge factor for better inclusion</li><li>H –––––––– true  # include bonds with hydrogen atoms</li><li>disulfides –––- false # include disulfide bonds</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/bonds.jl#L152-L161">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.distancebonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom" href="#BioMakie.distancebonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractAtom"><code>BioMakie.distancebonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">distancebonds( atms::Vector{T} )</code></pre><p>Returns a matrix of all bonds in <code>atms</code>. </p><p><strong>Optional Arguments:</strong></p><ul><li>cutoff –––––- 1.9   # distance cutoff for bonds between heavy atoms</li><li>hydrogencutoff –- 1.14  # distance cutoff for bonds with hydrogen atoms</li><li>H –––––––– true  # include bonds with hydrogen atoms</li><li>disulfides –––- false # include disulfide bonds</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/bonds.jl#L9-L19">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.firstlabel-Tuple{Function}" href="#BioMakie.firstlabel-Tuple{Function}"><code>BioMakie.firstlabel</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">firstlabel( inspectorfunc::Function )</code></pre><p>Show an example of the inspector label function looks like. The position <code>p</code> will not be available to this function, so it will be set to <code>nothing</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L181-L186">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.firstlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Function" href="#BioMakie.firstlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Function"><code>BioMakie.firstlabel</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">firstlabel( inspectorfunc::Observable{T} )</code></pre><p>Show an example of the inspector label function looks like. The position <code>p</code> will not be available to this function, so it will be set to <code>nothing</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L170-L175">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.getbonds-Tuple{BioStructures.Chain, Vararg{Any}}" href="#BioMakie.getbonds-Tuple{BioStructures.Chain, Vararg{Any}}"><code>BioMakie.getbonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">getbonds( chn::BioStructures.Chain )</code></pre><p>Returns a matrix of all bonds in <code>atms</code>. </p><p><strong>Optional Arguments:</strong></p><ul><li>algo ––––––- :knowledgebased 	# (:distance, :covalent) algorithm to find bonds</li><li>H –––––––– true				# include bonds with hydrogen atoms</li><li>cutoff –––––- 1.9				# distance cutoff for bonds between heavy atoms</li><li>extradistance –– 0.14				# fudge factor for better inclusion</li><li>disulfides –––- false				# include disulfide bonds</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/bonds.jl#L394-L405">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.getbonds-Tuple{Vector{MIToS.PDB.PDBResidue}}" href="#BioMakie.getbonds-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.getbonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">getbonds( resz::Vector{MIToS.PDB.PDBResidue} )</code></pre><p>Returns a matrix of all bonds in <code>resz</code>. </p><p><strong>Optional Arguments:</strong></p><ul><li>algo ––––––- :knowledgebased 	# (:distance, :covalent) algorithm to find bonds</li><li>H –––––––– true				# include bonds with hydrogen atoms</li><li>cutoff –––––- 1.9				# distance cutoff for bonds between heavy atoms</li><li>extradistance –– 0.14				# fudge factor for better inclusion</li><li>disulfides –––- false				# include disulfide bonds</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/bonds.jl#L478-L489">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.getbonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractResidue" href="#BioMakie.getbonds-Union{Tuple{Vector{T}}, Tuple{T}} where T&lt;:BioStructures.AbstractResidue"><code>BioMakie.getbonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">getbonds( resz::Vector{BioStructures.AbstractResidue} )</code></pre><p>Returns a matrix of all intra-residue bonds in <code>resz</code>.  Utilizes <code>sidechainbonds</code> to get bonds within each residue.</p><p><strong>Optional Arguments:</strong></p><ul><li>kwargs...			# passed to <code>sidechainbonds</code></li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/bonds.jl#L561-L569">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.inspectorlabel-Tuple{BioStructures.StructuralElementOrList}" href="#BioMakie.inspectorlabel-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.inspectorlabel</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">inspectorlabel( struc::BioStructures.StructuralElementOrList )</code></pre><p>Get the inspector label function for plotting a BioStructures.StructuralElementOrList. (ProteinStructure, Model, Chain, Residue, Atom, and lists of these)</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L90-L95">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.inspectorlabel-Tuple{Vector{MIToS.PDB.PDBAtom}}" href="#BioMakie.inspectorlabel-Tuple{Vector{MIToS.PDB.PDBAtom}}"><code>BioMakie.inspectorlabel</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">inspectorlabel( atms::Vector{MIToS.PDB.PDBAtom} )</code></pre><p>Get the inspector label function for plotting a vector of MIToS.PDB.PDBAtoms.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L118-L122">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.inspectorlabel-Tuple{Vector{MIToS.PDB.PDBResidue}}" href="#BioMakie.inspectorlabel-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.inspectorlabel</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">inspectorlabel( resz::Vector{MIToS.PDB.PDBResidue} )</code></pre><p>Get the inspector label function for plotting a Vector of MIToS.PDB.PDBResidues.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L105-L109">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.inspectorlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList" href="#BioMakie.inspectorlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList"><code>BioMakie.inspectorlabel</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">inspectorlabel( struc::Observable{T} ) where {T&lt;:BioStructures.StructuralElementOrList}</code></pre><p>Get the inspector label function for plotting a BioStructures StructuralElementOrList. (ProteinStructure, Model, Chain, Residue, Atom, and lists of these)</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L130-L135">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.inspectorlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}" href="#BioMakie.inspectorlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}"><code>BioMakie.inspectorlabel</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">inspectorlabel( atms::Observable{T} ) where {T&lt;:Vector{MIToS.PDB.PDBAtom}}</code></pre><p>Get the inspector label function for plotting a vector of MIToS.PDB.PDBAtoms.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L158-L162">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.inspectorlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}" href="#BioMakie.inspectorlabel-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}"><code>BioMakie.inspectorlabel</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">inspectorlabel( resz::Observable{T} ) where {T&lt;:Vector{MIToS.PDB.PDBResidue}}</code></pre><p>Get the inspector label function for plotting a vector of MIToS.PDB.PDBResidues.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L145-L149">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.msavalues" href="#BioMakie.msavalues"><code>BioMakie.msavalues</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">msavalues( msa::AbstractMatrix, resdict::AbstractDict )::Matrix{Real}</code></pre><p>Returns a matrix of numbers according to the given dictionary, where keys are residue letters and values are numbers. This matrix is used as input for <code>plotmsa</code> for the heatmap colors.</p><p>Default values for residue letters are from Kidera Factor values.  kf 2 is Kidera Factor 2 (size/volume-related). The KF dictionary is in <code>utils.jl</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/msa.jl#L55-L64">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plotmsa!" href="#BioMakie.plotmsa!"><code>BioMakie.plotmsa!</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">plotmsa!( fig, msa, msavalues, xlabels, ylabels )</code></pre><p>Plot a multiple sequence alignment (MSA) into a Figure. </p><p><strong>Example</strong></p><pre><code class="language-julia hljs">fig = Figure(resolution = (1100, 400))

plotmsa!( fig::Figure, msamatrix::Matrix{String}, matrixvals::Matrix{Float32},
			xlabels::Vector{String}, 	
			ylabels::Vector{String};
			kwargs... )</code></pre><p><strong>Optional Arguments:</strong></p><ul><li>xlabels –––- {1:height}</li><li>ylabels –––- {1:width}</li><li>sheetsize ––- [40,20]</li><li>gridposition – (1,1)</li><li>markersize –– 12</li><li>colorscheme –- :buda</li><li>markercolor –- :black</li><li>kwargs...   					# forwarded to scatter plot</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/msa.jl#L77-L101">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plotmsa!-NTuple{4, Any}" href="#BioMakie.plotmsa!-NTuple{4, Any}"><code>BioMakie.plotmsa!</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotmsa!( msamatrix, matrixvals, xlabels, ylabels )</code></pre><p>Plot a multiple sequence alignment (MSA) into a Figure. </p><p><strong>Example</strong></p><pre><code class="language-julia hljs">plotmsa!( msamatrix::Matrix{String}, matrixvals::Matrix{Float32},
			xlabels::Vector{String}, 	
			ylabels::Vector{String};
			kwargs... )</code></pre><p><strong>Optional Arguments:</strong></p><ul><li>xlabels –––- {1:height}</li><li>ylabels –––- {1:width}</li><li>sheetsize ––- [40,20]</li><li>gridposition – (1,1)</li><li>markersize –– 12</li><li>colorscheme –- :buda</li><li>markercolor –- :black</li><li>kwargs...   					# forwarded to scatter plot</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/msa.jl#L238-L260">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plotmsa!-NTuple{5, Any}" href="#BioMakie.plotmsa!-NTuple{5, Any}"><code>BioMakie.plotmsa!</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotmsa!( fig, msamatrix, matrixvals, xlabels, ylabels )</code></pre><p>Plot a multiple sequence alignment (MSA) into a Figure. </p><p><strong>Example</strong></p><pre><code class="language-julia hljs">fig = Figure(resolution = (1100, 400))

plotmsa!( fig::Figure, msamatrix::Matrix{String}, matrixvals::Matrix{Float32},
			xlabels::Vector{String}, 	
			ylabels::Vector{String};
			kwargs... )</code></pre><p><strong>Optional Arguments:</strong></p><ul><li>xlabels –––- {1:height}</li><li>ylabels –––- {1:width}</li><li>sheetsize ––- [40,20]</li><li>gridposition – (1,1)</li><li>markersize –– 12</li><li>colorscheme –- :buda</li><li>markercolor –- :black</li><li>kwargs...   					# forwarded to scatter plot</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/msa.jl#L196-L220">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plotmsa-NTuple{4, Any}" href="#BioMakie.plotmsa-NTuple{4, Any}"><code>BioMakie.plotmsa</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotmsa( fig, msamatrix, msavalues, xlabels, ylabels )</code></pre><p>Plot a multiple sequence alignment (MSA) on a Figure. </p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/msa.jl#L346-L351">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plotmsa-Tuple{Any, Any}" href="#BioMakie.plotmsa-Tuple{Any, Any}"><code>BioMakie.plotmsa</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotmsa( msa, msavalues )</code></pre><p>Plot a multiple sequence alignment (MSA). Returns a Figure, or a Figure and Observables for interaction.</p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">using MIToS.MSA
downloadpfam(&quot;PF00062&quot;)
msa = MIToS.MSA.read(&quot;PF00062.stockholm.gz&quot;, Stockholm, 
					generatemapping =true, useidcoordinates=true)
msamatrix, xlabels, ylabels = plottingdata(msa) .|&gt; Observable			
matrixvals = msavalues(msamatrix[]) |&gt; Observable

plotmsa( msa, matrixvals; kwargs... )</code></pre><p><strong>Optional Arguments:</strong></p><ul><li>resolution –––– (1100, 400)</li><li>sheetsize ––––- [40,20]</li><li>gridposition ––– (1,1)</li><li>colorscheme –––- :viridis</li><li>kwargs...    						# forwarded to scatter plot</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/msa.jl#L316-L340">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plotmsa-Tuple{Any}" href="#BioMakie.plotmsa-Tuple{Any}"><code>BioMakie.plotmsa</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotmsa( msa )</code></pre><p>Plot a multiple sequence alignment (MSA). Returns a Figure, or a Figure and Observables for interaction. </p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">downloadpfam(&quot;PF00062&quot;)
msa = MIToS.MSA.read(&quot;PF00062.stockholm.gz&quot;, Stockholm, 
					generatemapping =true, useidcoordinates=true)

plotmsa( msa; kwargs... )</code></pre><p><strong>Optional Arguments:</strong></p><ul><li>resolution –––– (1100, 400)</li><li>sheetsize ––––- [40,20]</li><li>gridposition ––– (1,1)</li><li>colorscheme –––- :viridis</li><li>resdict –––––- kideradict    # Dictionary of values (::Dict{String,Float}, &quot;Y&quot; =&gt; 1.48) for heatmap.</li><li>kf –––––––– 2             # If resdict == kideradict, this is the Kidera Factor. KF2 is size/volume-related.</li><li>kwargs...    						# forwarded to scatter plot</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/msa.jl#L280-L303">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plotstruc!-Tuple{Any, Any}" href="#BioMakie.plotstruc!-Tuple{Any, Any}"><code>BioMakie.plotstruc!</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotstruc!( fig, structure )</code></pre><p>Plot a protein structure(/chain/residues/atoms) into a Figure. </p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">fig = Figure()

using MIToS.PDB
pdbfile = MIToS.PDB.downloadpdb(&quot;2HHB&quot;)
struc = MIToS.PDB.read(pdbfile, PDBML) |&gt; Observable
strucplot = plotstruc!(fig, struc)

chain_A = pdb = @residues struc model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All
strucplot = plotstruc!(fig, chain_A)

chnatms = @atoms res_2hhb model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All atom All
strucplot = plotstruc!(fig, chnatms)

using BioStructures
struc = retrievepdb(&quot;2vb1&quot;, dir = &quot;data/&quot;) |&gt; Observable
strucplot = plotstruc!(fig, struc)

struc = read(&quot;data/2vb1_mutant1.pdb&quot;, BioStructures.PDB) |&gt; Observable
strucplot = plotstruc!(fig, struc)

chain_A = retrievepdb(&quot;2hhb&quot;, dir = &quot;data/&quot;)[&quot;A&quot;] |&gt; Observable
strucplot = plotstruc!(fig, chain_A)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>resolution ––- (800,600)</li><li>gridposition –- (1,1)  # if an MSA is already plotted, (2,1:3) works well</li><li>plottype –––- :covalent, :ballandstick, or :spacefilling</li><li>atomcolors ––- elecolors, others in <code>getbiocolors</code>, or provide a dict for atoms/residues like: &quot;N&quot; =&gt; :blue</li><li>markersize ––- 0.0</li><li>markerscale –– 1.0</li><li>bondtype –––- :knowledgebased, :covalent, or :distance</li><li>distance –––- 1.9  # distance cutoff for covalent bonds</li><li>inspectorlabel - :default, or define your own function like: (self, i, p) -&gt; &quot;atom: ... coords: ...&quot;</li><li>kwargs... ––– keyword arguments passed to the atom <code>meshscatter</code></li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L760-L802">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plotstruc!-Tuple{Figure, Observable}" href="#BioMakie.plotstruc!-Tuple{Figure, Observable}"><code>BioMakie.plotstruc!</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotstruc!( fig, structure )</code></pre><p>Plot a protein structure(/chain/residues/atoms) into a Figure. </p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">fig = Figure()

using MIToS.PDB
pdbfile = MIToS.PDB.downloadpdb(&quot;2HHB&quot;)
struc = MIToS.PDB.read(pdbfile, PDBML) |&gt; Observable
strucplot = plotstruc!(fig, struc)

chain_A = pdb = @residues struc model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All
strucplot = plotstruc!(fig, chain_A)

chnatms = @atoms res_2hhb model &quot;1&quot; chain &quot;A&quot; group &quot;ATOM&quot; residue All atom All
strucplot = plotstruc!(fig, chnatms)

using BioStructures
struc = retrievepdb(&quot;2vb1&quot;, dir = &quot;data/&quot;) |&gt; Observable
strucplot = plotstruc!(fig, struc)

struc = read(&quot;data/2vb1_mutant1.pdb&quot;, BioStructures.PDB) |&gt; Observable
strucplot = plotstruc!(fig, struc)

chain_A = retrievepdb(&quot;2hhb&quot;, dir = &quot;data/&quot;)[&quot;A&quot;] |&gt; Observable
strucplot = plotstruc!(fig, chain_A)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>resolution ––- (800,600)</li><li>gridposition –- (1,1)  # if an MSA is already plotted, (2,1:3) works well</li><li>plottype –––- :covalent, :ballandstick, or :spacefilling</li><li>atomcolors ––- elecolors, others in <code>getbiocolors</code>, or provide a dict for atoms/residues like: &quot;N&quot; =&gt; :blue</li><li>markersize ––- 0.0</li><li>markerscale –– 1.0</li><li>bondtype –––- :knowledgebased, :covalent, or :distance</li><li>distance –––- 1.9  # distance cutoff for covalent bonds</li><li>inspectorlabel - :default, or define your own function like: (self, i, p) -&gt; &quot;atom: ... coords: ...&quot;</li><li>kwargs... ––– keyword arguments passed to the atom <code>meshscatter</code></li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L658-L700">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plotstruc-Tuple{BioStructures.StructuralElementOrList}" href="#BioMakie.plotstruc-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.plotstruc</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotstruc( struc::BioStructures.StructuralElementOrList )</code></pre><p>Create and return a Makie Figure for a protein structural element. </p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">using BioStructures
struc = retrievepdb(&quot;2vb1&quot;, dir = &quot;data/&quot;) |&gt; Observable
sv = plotstruc(struc)

struc = read(&quot;data/2vb1_mutant1.pdb&quot;, BioStructures.PDB) |&gt; Observable
sv = plotstruc(struc)</code></pre><p><strong>Optional Arguments:</strong></p><ul><li>resolution ––- (800,600)</li><li>gridposition –- (1,1)  # if an MSA is already plotted, (2,1:3) works well</li><li>plottype –––- :covalent, :ballandstick, or :spacefilling</li><li>atomcolors ––- elecolors, others in <code>getbiocolors</code>, or provide a dict for atoms/residues like: &quot;N&quot; =&gt; :blue</li><li>markersize ––- 0.0</li><li>markerscale –– 1.0</li><li>bondtype –––- :knowledgebased, :covalent, or :distance</li><li>distance –––- 1.9  # distance cutoff for covalent bonds</li><li>inspectorlabel - :default, or define your own function like: (self, i, p) -&gt; &quot;atom: ... coords: ...&quot;</li><li>kwargs... ––– keyword arguments passed to the atom <code>meshscatter</code></li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L810-L836">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plotstruc-Tuple{Observable}" href="#BioMakie.plotstruc-Tuple{Observable}"><code>BioMakie.plotstruc</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plotstruc( struc::Observable )</code></pre><p>Create and return a Makie Figure for a protein structural element. </p><p><strong>Examples</strong></p><pre><code class="language-julia hljs">using BioStructures
struc = retrievepdb(&quot;2vb1&quot;, dir = &quot;data/&quot;) |&gt; Observable
sv = plotstruc(struc)

struc = read(&quot;data/2vb1_mutant1.pdb&quot;, BioStructures.PDB) |&gt; Observable
sv = plotstruc(struc)</code></pre><p><strong>Optional Arguments:</strong></p><ul><li>resolution ––- (800,600)</li><li>gridposition –- (1,1)  # if an MSA is already plotted, (2,1:3) works well</li><li>plottype –––- :covalent, :ballandstick, or :spacefilling</li><li>atomcolors ––- elecolors, others in <code>getbiocolors</code>, or provide a dict for atoms/residues like: &quot;N&quot; =&gt; :blue</li><li>markersize ––- 0.0</li><li>markerscale –– 1.0</li><li>bondtype –––- :knowledgebased, :covalent, or :distance</li><li>distance –––- 1.9  # distance cutoff for covalent bonds</li><li>inspectorlabel - :default, or define your own function like: (self, i, p) -&gt; &quot;atom: ... coords: ...&quot;</li><li>kwargs... ––– keyword arguments passed to the atom <code>meshscatter</code></li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L842-L868">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plottingdata-Tuple{BioStructures.StructuralElementOrList}" href="#BioMakie.plottingdata-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.plottingdata</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plottingdata( struc::BioStructures.StructuralElementOrList )</code></pre><p>Collects data for plotting from a BioStructures.StructuralElementOrList. This function returns an OrderedDict of Observables that are the main data used for plotting, for ease of use and consistency. </p><pre><code class="nohighlight hljs">OrderedDict(&quot;atoms&quot; =&gt; ..., 
            &quot;coords&quot; =&gt; ..., 
            &quot;colors&quot; =&gt; ...,
            &quot;sizes&quot; =&gt; ...,
            &quot;bonds&quot; =&gt; ...)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>colors –––- elecolors | Options - elecolors, aquacolors, shapelycolors, maecolors</li><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L438-L454">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plottingdata-Tuple{MIToS.MSA.AbstractMultipleSequenceAlignment}" href="#BioMakie.plottingdata-Tuple{MIToS.MSA.AbstractMultipleSequenceAlignment}"><code>BioMakie.plottingdata</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plottingdata( msa::AbstractMultipleSequenceAlignment )</code></pre><p>Collects data for plotting (residue string matrix, x labels, and y labels).</p><p>This function converts an AbstractMultipleSequenceAlignment (from MIToS.MSA)  to a matrix of residue characters, x labels, and y labels.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/msa.jl#L6-L13">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plottingdata-Tuple{Vector{FASTX.FASTA.Record}}" href="#BioMakie.plottingdata-Tuple{Vector{FASTX.FASTA.Record}}"><code>BioMakie.plottingdata</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plottingdata( msa::Vector{FASTX.FASTA.Record} )</code></pre><p>Collects data for plotting (residue string matrix, x labels, and y labels).</p><p>This function converts a Vector{FASTX.FASTA.Record}  to a matrix of residue characters, x labels, and y labels.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/msa.jl#L38-L45">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plottingdata-Tuple{Vector{MIToS.PDB.PDBAtom}}" href="#BioMakie.plottingdata-Tuple{Vector{MIToS.PDB.PDBAtom}}"><code>BioMakie.plottingdata</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plottingdata( atms::Vector{MIToS.PDB.PDBAtom} )</code></pre><p>Collects data for plotting from a Vector of MIToS.PDB.PDBAtoms. This function returns an OrderedDict of Observables that are the main data used for plotting, for ease of use and consistency. </p><pre><code class="nohighlight hljs">OrderedDict(&quot;atoms&quot; =&gt; ..., 
            &quot;coords&quot; =&gt; ..., 
            &quot;colors&quot; =&gt; ...,
            &quot;sizes&quot; =&gt; ...,
            &quot;bonds&quot; =&gt; nothing)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>colors –––- elecolors | Options - elecolors, aquacolors, shapelycolors, maecolors</li><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L517-L533">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plottingdata-Tuple{Vector{MIToS.PDB.PDBResidue}}" href="#BioMakie.plottingdata-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.plottingdata</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plottingdata( resz::Vector{MIToS.PDB.PDBResidue} )</code></pre><p>Collects data for plotting from a Vector of MIToS.PDB.PDBResidues. This function returns an OrderedDict of Observables that are the main data used for plotting, for ease of use and consistency. </p><pre><code class="nohighlight hljs">OrderedDict(&quot;atoms&quot; =&gt; ..., 
            &quot;coords&quot; =&gt; ..., 
            &quot;colors&quot; =&gt; ...,
            &quot;sizes&quot; =&gt; ...,
            &quot;bonds&quot; =&gt; ...)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>colors –––- elecolors | Options - elecolors, aquacolors, shapelycolors, maecolors</li><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L477-L493">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plottingdata-Tuple{Vector{Tuple{String, String}}}" href="#BioMakie.plottingdata-Tuple{Vector{Tuple{String, String}}}"><code>BioMakie.plottingdata</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plottingdata( msa::Vector{Tuple{String,String}} )</code></pre><p>Collects data for plotting (residue string matrix, x labels, and y labels).</p><p>This function converts a Vector{Tuple{String,String}} (from FastaIO) to a matrix of residue characters, x labels, and y labels.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/msa.jl#L21-L28">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plottingdata-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList" href="#BioMakie.plottingdata-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList"><code>BioMakie.plottingdata</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plottingdata( struc::Observable{T} ) where {T&lt;:BioStructures.StructuralElementOrList}</code></pre><p>Collects data for plotting from a BioStructures.StructuralElementOrList. This function returns an OrderedDict of Observables that are the main data used for plotting, for ease of use and consistency. </p><pre><code class="nohighlight hljs">OrderedDict(&quot;atoms&quot; =&gt; ..., 
            &quot;coords&quot; =&gt; ..., 
            &quot;colors&quot; =&gt; ...,
            &quot;sizes&quot; =&gt; ...,
            &quot;bonds&quot; =&gt; ...)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>colors –––- elecolors | Options - elecolors, aquacolors, shapelycolors, maecolors</li><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L550-L566">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plottingdata-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}" href="#BioMakie.plottingdata-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBAtom}"><code>BioMakie.plottingdata</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plottingdata( atms::Observable{T} ) where {T&lt;:Vector{MIToS.PDB.PDBAtom}}</code></pre><p>Collects data for plotting from a Vector of MIToS.PDB.PDBAtoms. This function returns an OrderedDict of Observables that are the main data used for plotting, for ease of use and consistency. </p><pre><code class="nohighlight hljs">OrderedDict(&quot;atoms&quot; =&gt; ..., 
            &quot;coords&quot; =&gt; ..., 
            &quot;colors&quot; =&gt; ...,
            &quot;sizes&quot; =&gt; ...,
            &quot;bonds&quot; =&gt; nothing)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>colors –––- elecolors | Options - elecolors, aquacolors, shapelycolors, maecolors</li><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L626-L642">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.plottingdata-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}" href="#BioMakie.plottingdata-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}"><code>BioMakie.plottingdata</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">plottingdata( resz::Observable{T} ) where {T&lt;:Vector{MIToS.PDB.PDBResidue}}</code></pre><p>Collects data for plotting from a Vector of MIToS.PDB.PDBResidues. This function returns an OrderedDict of Observables that are the main data used for plotting, for ease of use and consistency. </p><pre><code class="nohighlight hljs">OrderedDict(&quot;atoms&quot; =&gt; ..., 
            &quot;coords&quot; =&gt; ..., 
            &quot;colors&quot; =&gt; ...,
            &quot;sizes&quot; =&gt; ...,
            &quot;bonds&quot; =&gt; ...)</code></pre><p><strong>Keyword Arguments:</strong></p><ul><li>colors –––- elecolors | Options - elecolors, aquacolors, shapelycolors, maecolors</li><li>radiustype –- :covalent | Options - :cov, :covalent, :vdw, :vanderwaals, :bas, :ballandstick</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L588-L604">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.rescolors-Tuple{BioStructures.StructuralElementOrList}" href="#BioMakie.rescolors-Tuple{BioStructures.StructuralElementOrList}"><code>BioMakie.rescolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">rescolors( struc::BioStructures.StructuralElementOrList )</code></pre><p>Get a Vector of colors for the atoms from a BioStructures.StructuralElementOrList. To see all default element and amino acid colorschemes, use the <code>getbiocolors()</code> function. Keyword argument <code>colors</code> takes a Dict which maps residue to color. (&quot;C&quot; =&gt; :red)</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors, shapelycolors, maecolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L286-L295">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.rescolors-Tuple{Vector{MIToS.PDB.PDBResidue}}" href="#BioMakie.rescolors-Tuple{Vector{MIToS.PDB.PDBResidue}}"><code>BioMakie.rescolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">rescolors( resz::Vector{MIToS.PDB.PDBResidue} )</code></pre><p>Get a Vector of colors for the atoms from a Vector of MIToS.PDB.PDBResidues. To see all default element and amino acid colorschemes, use the <code>getbiocolors()</code> function. Keyword argument <code>colors</code> takes a Dict which maps residue to color. (&quot;C&quot; =&gt; :red)</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors, shapelycolors, maecolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L303-L312">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.rescolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList" href="#BioMakie.rescolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:BioStructures.StructuralElementOrList"><code>BioMakie.rescolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">rescolors( struc::Observable{T} ) where {T&lt;:BioStructures.StructuralElementOrList}</code></pre><p>Get a Vector of colors for the atoms from a BioStructures.StructuralElementOrList. To see all default element and amino acid colorschemes, use the <code>getbiocolors()</code> function. Keyword argument <code>colors</code> takes a Dict which maps residue to color. (&quot;C&quot; =&gt; :red)</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors, shapelycolors, maecolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L321-L330">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.rescolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}" href="#BioMakie.rescolors-Union{Tuple{Observable{T}}, Tuple{T}} where T&lt;:Vector{MIToS.PDB.PDBResidue}"><code>BioMakie.rescolors</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">rescolors( resz::Observable{T} ) where {T&lt;:Vector{MIToS.PDB.PDBResidue}}</code></pre><p>Get a Vector of colors for the atoms from a Vector of MIToS.PDB.PDBResidues. To see all default element and amino acid colorschemes, use the <code>getbiocolors()</code> function. Keyword argument <code>colors</code> takes a Dict which maps residue to color. (&quot;C&quot; =&gt; :red)</p><p><strong>Keyword Arguments:</strong></p><ul><li>colors –- elecolors | Options - elecolors, aquacolors, shapelycolors, maecolors</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/structure.jl#L338-L347">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="BioMakie.sidechainbonds-Tuple{BioStructures.AbstractResidue, Vararg{Any}}" href="#BioMakie.sidechainbonds-Tuple{BioStructures.AbstractResidue, Vararg{Any}}"><code>BioMakie.sidechainbonds</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">sidechainbonds( res::BioStructures.AbstractResidue, selectors... )</code></pre><p>Returns a matrix of sidechain bonds in <code>res</code>. </p><p><strong>Optional Arguments:</strong></p><ul><li>algo ––––––- :knowledgebased 	# (:distance, :covalent) algorithm to find bonds</li><li>H –––––––– true				# include bonds with hydrogen atoms</li><li>cutoff –––––- 1.9				# distance cutoff for bonds between heavy atoms</li><li>extradistance –– 0.14				# fudge factor for better inclusion</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/kool7d/BioMakie.jl/blob/d648777622ddffb0d9b0995f56e2be215b047911/src/bonds.jl#L286-L296">source</a></section></article></article><nav class="docs-footer"><a class="docs-footer-prevpage" href="../">« Home</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 0.28.0-DEV on <span class="colophon-date" title="Friday 17 February 2023 09:50">Friday 17 February 2023</span>. Using Julia version 1.8.5.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
